kmultiversx/src/kmultiversx/kasmer.py

```diff
@@ -166,12 +200,15 @@ def run_test(krun: KRun, sym_conf: KInner, init_subst: dict[str, KInner], endpoi
     }
     tx_steps = KSequence([set_exit_code(1)] + get_steps_sc_call(step) + [set_exit_code(0)])
 
-    subst = init_subst.copy()
-    subst['K_CELL'] = tx_steps
-    conf_with_steps = Subst(subst)(sym_conf)
+    def subst_k_cell(p: Pattern) -> Pattern:
+        if p == K_STEPS_VAR_KORE:
+            return krun.kast_to_kore(tx_steps)
+        return p
+
+    test_conf = sym_conf.top_down(subst_k_cell)
 
     try:
-        run_config_and_check_empty(krun, conf_with_steps, pipe_stderr=True)
+        run_pattern_and_check_exit_code(krun, test_conf, pipe_stderr=True)
```

Comment:
Consider moving the `try` block into `run_pattern_and_check_exit_code`.

Reply:
Btw, how much of the input to `interpreter` comes from the user? Maybe the exception should be translated to `AssertionError` instead.